### ISSUE TITLE
Get output after closing and reopening the browser tab with notebook

### ISSUE THREAD

**Issue by VladimirVisnovsky** [jupyterlab/jupyterlab#12422]
### Problem
I have a following situation. There is a multiple hour computation ran by Jupyter notebook. After the computation started I would like to close the tab. Then open the tab again (maybe on different device if the computation was run in the cloud) and check the output. But after I reopen the tab there is no output since the mappings between "Execution request" and the "Execution reply" are gone. There is a [stackoverflow thread](https://stackoverflow.com/questions/45789906/can-i-autosave-a-running-jupyter-python-notebook-without-having-it-open-in-a-bro) describing the problem in more precise way so I don't have to duplicate the concrete issue description. 

### Proposed Solution
What I would expect to see is that after starting of computation I would not have to keep my notebook opened in browser to keep the mappings between "Execution request" and the "Execution reply" alive. Instead I could close the tab and after multiple hours just return to the notebook for the output.

Could the notebook model be moved from client side to the server side? This way the mapping between "Execution request" and the "Execution reply" would not be lost and there would be no problem if one closed the tab, then reopened it and wanted to check the output.

### Additional context
If I understand correctly this issue is known by the Jupyter team and I would like to get an update in case it has been forgotten since it would resolve my ongoing problem. Also I would appreciate some kind of workaround suggestion. I am printing everything twice into the standard output and into the log file (where I can tail the appending output real-time). But this is not a very elegant solution.

**Comment by jweill-aws** [comment]
Elyra allows for the running of a notebook as a batch job, which may address the use case that you're describing:
https://elyra.readthedocs.io/en/latest/getting_started/overview.html#ability-to-run-a-notebook-python-or-r-script-as-a-batch-job

I don't see any other issues in JupyterLab to allow the server to both execute the code _and_ modify the notebook, so that you can (for example) close the tab or open your notebook on a separate instance and see the results of long-running scripts. For this reason, I've accepted this issue.

**Comment by jweill-aws** [comment]
As a workaround, you can run a computation and save the output to a variable. Then, if you close your browser tab and execution continues, you can come back, and the output should be accessible as a variable.

**Comment by chenmicheal** [comment]
> 

Hi @jweill-aws , could you tell me how to save the output to a variable? 
thank you in advance!

**Comment by jweill-aws** [comment]
You can save the output by assigning it to a variable using standard Python syntax. If you still want to display it, you can refer to the variable on the next line.

For example, change
``` python
long_running_function(variables)
```

to:
``` python
myoutput = long_running_function(variables)
myoutput
```

**Comment by VladimirVisnovsky** [comment]
Also @chenmicheal,  take into consideration magic command %%capture which (quote from documentation) "runs the cell, capturing stdout, stderr, and IPython’s rich display() calls." With this magic command you can save all output of a cell into the variable in a native way for Jupyter notebook

Source:
https://ipython.readthedocs.io/en/stable/interactive/magics.html

Usage examples:
https://notebook.community/lifeinoppo/littlefishlet-scode/RES/REF/python_sourcecode/ipython-master/examples/IPython%20Kernel/Capturing%20Output

**Comment by chenmicheal** [comment]
thank you both very much!

**Comment by lumbric** [comment]
This is an often requested feature and discussed at many places. I want to add some links here, maybe it helps to find stuff.

Workarounds and explanation:
- [a pretty good summary including workarounds on stackoverflow, last updated in 2017](https://stackoverflow.com/a/36845963/859591)
- [Blog post on towardsdatascience including workarounds](https://towardsdatascience.com/keep-jupyter-notebook-running-even-after-browser-is-closed-9a1937b7c615)

Other Github issues:
- ipython/ipython#4140 
- jupyter/notebook#1647
- jupyter/notebook#641
- jupyter/notebook#810

[Zeppelin](https://zeppelin.apache.org/) and [Datalore](datalore.jetbrains.com/) have such a feature implemented.

[A comment](https://github.com/jupyter/notebook/issues/641#issuecomment-709569918) suggests that this issue might be related to [RTC](https://jupyterlab.readthedocs.io/en/stable/user/rtc.html), but JupyterLab 3.4.4 does not support reconnecting after all browser tabs have been closed even with RTC enabled (just tested it).

**Comment by krassowski** [comment]
Closing as a duplicate of #2833.

Of note there is now a PR implementing this: https://github.com/jupyterlab/jupyterlab/pull/15448

> If I understand correctly this issue is known by the Jupyter team and I would like to get an update in case it has been forgotten since it would resolve my ongoing problem

Just FYI, Jupyter is open-source, community-driven everyone is welcome to contribute one way or the other. If you would would like to discuss a particular topic, [weekly jupyterlab meetings](https://github.com/jupyterlab/team-compass#weekly-jupyterlab-team-meeting) may be suitable to get an informal answer but it is unlikely any one contributor would speak for the entire community on such high-level feature requests.
